cnd/src/dependencies.rs

```diff
@@ -0,0 +1,25 @@
+use crate::{
+    seed::{Seed, SwapSeed},
+    swap_protocols::{
+        metadata_store::InMemoryMetadataStore, rfc003::state_store::InMemoryStateStore,
+        LedgerConnectors, SwapId,
+    },
+};
+use std::sync::Arc;
+
+/// Core cnd dependencies required by both the libp2p network layer and the HTTP
+/// API layer.
+#[allow(missing_debug_implementations)]
+#[derive(Clone)]
+pub struct Dependencies {
```

Comment:
Personal opinion ahead: I think you are following DRY a bit to rigorous here :see_no_evil: 
Just because two structs have the same fields, doesn't mean it is (bad) duplication. This kind of de-duplication often leads to dependencies between modules that can actually bite you fairly badly. Plus, it is a layer of indirection that readers of the code will need to follow.

I you feel strongly about it, I am happy to keep it and see how it goes :)

Reply:
Interesting point, your view is somewhat expressed by the way I declared the second instance of `Dependencies` in main explicitly instead of cloning the first instance passed into the `ComitNode`
```
    let http_api_dependencies = http_api::Dependencies {
        dependencies: Arc::new(Dependencies {
            ledger_events: ledger_events.clone(),
            metadata_store: Arc::clone(&metadata_store),
            state_store: Arc::clone(&state_store),
            seed,
        }),
        swarm: Arc::clone(&swarm),
    };
```

I actually tried putting the `Dependencies` fields into a struct that was used by the http_api layer but this did not work out as nice - open to being corrected but it seems we need _some_ object to implement `Spawn` on.  `spawn()` needs access to the seed and the `ledger_events` so AFAICT we need an object that holds both of these at least.  Am I missing someting?  Actually the seed could be passed into spawn then spawn could be implemented on `LedgerConnectors` but the next developer is going to struggle to see why that is the case?  I can have a play with this and see if I come up with something.